### PR TITLE
Allow returning Resource from handle_errors

### PR DESCRIPTION
**Pull request by IlyaSkriblovsky**
Currently it is not permitted to return `Resource` instances from `Klein.handle_errors()` which seems inconsistent.

This PR replaces duplicated handling of `IRenderable` in `handle_errors` with calling the same `process()` in `KleinResource` as for returned value from non-error route functions.

### CODE REVIEW

**Comment by wsanchez** [comment]
Did #139 already solve this?  If so, this fix is arguably simpler, so we should consider whether it's better.  Or am I confused and is this something different?

**Comment by IlyaSkriblovsky** [comment]
Seems like #139 is about handling `IRenderable` results, but this PR is about `Resource` results. `Resource` is not `twsted.web.iweb.IRenderable` despite the similar interface.
I've just checked, returning `Resource` from `handle_errors` still fails on `master` branch.

IMHO this fix looks simplier than #139 and reduces code duplication. But you are no doubt more expirienced in Klein internals, so please consider.

File: src/klein/test/test_resource.py [hunk 1]
```diff
@@ -869,6 +869,30 @@ def handle_not_found(request, failure):
         self.assertEqual(request.processingFailed.called, False)
         self.assertEqual(request.getWrittenData(), rendered)
 
+    def test_errorHandlerReturnsResource(self):
+        """
+        Resources returned by L{handle_errors} are rendered
```
Comment (glyph) [inline src/klein/test/test_resource.py]: I think you mean `L{Klein.handle_errors}` here?

Comment (IlyaSkriblovsky) [inline src/klein/test/test_resource.py]: Sure. Sorry, didn't have time to change it before you merged this PR, so created #188 for this :)

**Pull request merged**